src/main/java/ua/softserveinc/tc/service/impl/BaseServiceImpl.java

```diff
@@ -8,44 +8,48 @@
 
 import java.util.List;
 
-@Transactional
 public abstract class BaseServiceImpl<T> implements BaseService<T> {
 
     @Autowired
     private BaseDao<T> baseDao;
 
     @Override
+    @Transactional(readOnly = true)
+    public T findById(Object id) {
```

Comment:
I think we should consider to use Optional as a return type in all methods that can return null. This is a perfect use case for such situations.